### DISCUSSION TITLE
Import env. when importing compose stacks

### DISCUSSION THREAD

**Discussion by vlude** [moghtech/komodo#219] [Q&A] [answered]
I used the documentation to import my compose stacks which are currently (running&) and managed by Dockge. Importing the stacks works fine when pointing Komodo to the right folder. Except for the .env files. It does not import these, but only writes to them. Is it possible to import them in some way? Or do I need to go for the manual way: copying them over, file per file.

**Comment by mbecker20** [comment]
I see what you mean, it doesn't support importing from existing env unfortunately. This is possible for compose files in Files on Server mode because they aren't actually imported into the resource configuration, just piped through from your servers filesystem (with some caching in between). But yeah, to manage env in the UI as part of the Stack configuration, copy paste is necessary.

It would be cool for Komodo to see and manage any files related to some deployment, including .env files on the host. I opened this issue myself a while ago: https://github.com/mbecker20/komodo/issues/92

Also note that in this case, they don't *need* to be managed by Komodo, you leave it empty and it will auto pick up an existing `.env`, and if you have more you can use `Additional Env File` to point to them.

**Reply by vlude** [reply]
Hi Maxwell, thank you for the information (& making a great product). As I would prefer to have all the necessary files in one tool, I will use the copy paste method for now.
One additional thing I am thinking of now: configuring the compose files in together with variables in .env feels kind off right now. In comparison with Dockge, it is not possible to get both files in one screen. Would it be an idea to paste the .env contents below the compose files?

**Reply by mbecker20** [reply, marked as answer]
In Komodo, there is a clear split between the declared configuration (like env and other Komodo specific option that you give Komodo directly), and inferred configuration (pulling compose file from your server / repo).

The current implementation has env strictly part of the declared configuration, so its shown on the Config tab. If you use UI defined Stack, then the compose file is part of the declared Komodo configuration, and is also included in the Config tab. You can try this yourself, and see the compose file edit right above the environment configuration as you are looking for. I recommend going with UI defined Stacks where possible, as they are then managed with resource syncs directly, and in the end you have less files and less config sprawl.

However with Files on Server / Git repo, the compose file isn't actually part of the Komodo configuration, they exist externally. So it gets its own tab, the Info tab, to separate it from the true configuration. To me, this is in line with the reality of how Komodo is working with Stack in Files on Server / Git repo mode.
